src/Simplex/Chat/Library/Commands.hs

```diff
@@ -2023,75 +2023,171 @@ processChatCommand' vr = \case
           updateCIGroupInvitationStatus user g CIGISAccepted `catchChatError` (toView . CRChatError (Just user))
           pure $ CRUserAcceptedGroupSent user g {membership = membership {memberStatus = GSMemAccepted}} Nothing
         Nothing -> throwChatError $ CEContactNotActive ct
-  APIMemberRole groupId memberId memRole -> withUser $ \user -> do
-    Group gInfo@GroupInfo {membership} members <- withFastStore $ \db -> getGroup db vr user groupId
-    if memberId == groupMemberId' membership
-      then changeMemberRole user gInfo members membership $ SGEUserRole memRole
-      else case find ((== memberId) . groupMemberId') members of
-        Just m -> changeMemberRole user gInfo members m $ SGEMemberRole memberId (fromLocalProfile $ memberProfile m) memRole
-        _ -> throwChatError CEGroupMemberNotFound
+  APIMembersRole groupId memberIds newRole -> withUser $ \user ->
+    withGroupLock "memberRole" groupId . procCmd $ do
+      g@(Group gInfo members) <- withFastStore $ \db -> getGroup db vr user groupId
+      when (selfSelected gInfo) $ throwChatError $ CECommandError "can't change role for self"
```

Comment:
prohibited to change role for self to not further complicate processing. It's not used anywhere, only single test did it. In UI we don't even allow to interact with own member. If needed I think better to do as separate api